lte/gateway/c/core/oai/tasks/nas/api/mme/mme_api.c

```diff
@@ -446,118 +310,109 @@ status_code_e mme_api_new_guti(
   } else {
     OAILOG_FUNC_RETURN(LOG_NAS, RETURNerror);
   }
+  par_tai_list = emm_verify_orig_tai(*originating_tai);
+  if (par_tai_list == NULL) {
+    OAILOG_ERROR_UE(
+        LOG_NAS, imsi64,
+        "No matching partial list found for originating TAI!" TAI_FMT "\n",
+        TAI_ARG(originating_tai));
+    OAILOG_FUNC_RETURN(LOG_NAS, RETURNerror);
+  }
+  if (!par_tai_list->plmn) {
+    OAILOG_ERROR_UE(LOG_NAS, imsi64, "config PLMN is NULL\n");
+    OAILOG_FUNC_RETURN(LOG_NAS, RETURNerror);
+  }
+  if (!par_tai_list->tac) {
+    OAILOG_ERROR_UE(LOG_NAS, imsi64, "config TAC is NULL\n");
+    OAILOG_FUNC_RETURN(LOG_NAS, RETURNerror);
+  }
 
+  OAILOG_INFO(
+      LOG_NAS,
+      "Matching partial list for originating TAI found! typeOfList=%d\n",
+      par_tai_list->list_type);
   int j = 0;  // keeps track of number of partial lists with matching PLMN
```

Comment:
this is no longer needed as at this point we know that we know that we will return exactly 1 partial list that has a matching PLMN and TAC.